src/opm/input/eclipse/Schedule/Network/ExtNetwork.cpp

```diff
@@ -197,7 +197,12 @@ void ExtNetwork::update_node(Node node)
                                [&name](const Branch& b) { return b.uptree_node() == name || b.downtree_node() == name;});
 
     if (branch != this->m_branches.end() && branch->downtree_node() == name) {
-        if (node.as_choke() && branch->vfp_table().has_value())
+        bool node_is_wellgroup = false;
```

Comment:
Thanks for pointing out to this. I think I might read it wrongly, there is some confusions between standard network and extended network.

Reply:
Then I see because my understanding of this point looks like wrong. 

I have been thinking that only `well group` can be autochoke, then my suggestion is based on that.  Now I understand that why you did not having the `target_group` part not included in the `if (group.wellgroup())` . 

Then you need to revert to include the following part in the `if (group.wellgroup())` condition. 
```c++
            std::string target_group = name;
            const auto& target_item = record.getItem<ParserKeywords::NODEPROP::CHOKE_GROUP>();

            if (target_item.hasValue(0)) {
                target_group = target_item.get<std::string>(0);
            }

            if (target_group != name) {
                if (handlerContext.state().groups.has(name)) {
                    const auto& group = handlerContext.state().groups.get(name);
                    if (group.numWells() > 0) {
                        throw std::invalid_argument("A manifold group must respond to its own target");
                    }
                }
            }
```

They should be outside of the `if (group.wellgroup())` condition, because they might not be well group. 

At the same time, 
The following code, 
```c++
            if (target_group != name) {
                if (handlerContext.state().groups.has(name)) {
                    if (group.numWells() > 0) {
                        throw std::invalid_argument("A manifold group must respond to its own target");
                    }
                }
```

should be
```c++
   if (group.wellgroup() && target_group != name) {
        throw ...
```

Preferably, you can throw `OpmInputError`.

Reply:
Referring to our chat we keep  the `target_group` code within the if (`group.wellgroup()`) following your first suggestion. Changed throw to `OpmInputError`.